<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Legana Fingerhut" />


<title>Introduction to ampir</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to ampir</h1>
<h4 class="author">Legana Fingerhut</h4>



<p>The <strong>ampir</strong> (short for <strong>a</strong>nti<strong>m</strong>icrobial <strong>p</strong>eptide prediction <strong>i</strong>n <strong>r</strong> ) package was designed to be an open and user-friendly method to predict AMPs (antimicrobial peptides) from any given size protein dataset. <strong>ampir</strong> uses four sequential functions to create a four step approach to predict AMPs.</p>
<div id="brief-background" class="section level1">
<h1>Brief background</h1>
<p><strong>ampir</strong> uses a <em>supervised statistical machine learning</em> approach to predict AMPs. Basically this involves making a statistical model based on <em>input</em> data to predict <em>output</em> data <a href="http://www-bcf.usc.edu/~gareth/ISL/">James, Witten, Hastie &amp; Tibshirani 2013</a>. The input data are also known as <em>features</em> which are used to describe the data. To predict AMPs, physicochemical and compositional properties of protein sequences are used as features <a href="https://journal.r-project.org/archive/2015/RJ-2015-001/RJ-2015-001.pdf">Osorio, Rondón-Villarreal &amp; Torres 2015</a>. Therefore, within this package, it is important to <strong><em>follow an order of functions</em></strong>.</p>
<div id="order-of-functions-to-follow" class="section level2">
<h2>Order of functions to follow</h2>
<ol style="list-style-type: decimal">
<li><code>read_faa()</code> to read FASTA amino acid files.</li>
<li><code>remove_nonstandard_aa()</code> to remove non standard amino acid sequences.</li>
<li><code>calculate_features()</code> to calculate data descriptors used by the predictive model.</li>
<li><code>predict_AMP_prob()</code> to predict the AMP probability of a protein.</li>
</ol>
</div>
</div>
<div id="example-workflow" class="section level1">
<h1>Example workflow</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(ampir)</a></code></pre></div>
<div id="step-1-read-fasta-amino-acid-files-with-read_faa" class="section level3">
<h3>Step 1: Read FASTA amino acid files with <code>read_faa()</code></h3>
<p><code>read_faa()</code> reads FASTA amino acid files as a dataframe.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">my_protein &lt;-<span class="st"> </span><span class="kw">read_faa</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/bat_protein.fasta&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;ampir&quot;</span>))</a></code></pre></div>
<table>
<caption>My protein</caption>
<thead>
<tr class="header">
<th align="left">seq.name</th>
<th align="left">seq.aa</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G1P6H5_MYOLU</td>
<td align="left">MALTVRIQAACLLLLLLASLTSYSLLLSQTTQLAD</td>
</tr>
</tbody>
</table>
</div>
<div id="step-2-remove-non-standard-amino-acids-with-remove_nonstandard_aa" class="section level3">
<h3>Step 2: Remove non standard amino acids with <code>remove_nonstandard_aa()</code></h3>
<p><code>remove_nonstandard_aa()</code> is used to remove sequences that contain anything other than the 20 standard amino acids. These sequences are removed to circumvent potential complications with calculations.</p>
<table>
<caption>Example dataframe with a nonsense protein</caption>
<thead>
<tr class="header">
<th align="left">seq.name</th>
<th align="left">seq.aa</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G1P6H5_MYOLU</td>
<td align="left">MALTVRIQAACLLLLLLASLTSYSLLLSQTTQLAD</td>
</tr>
<tr class="even">
<td align="left">fake_sequence</td>
<td align="left">MKVTHEUSYR$GXMBIJIDG*M80-%</td>
</tr>
</tbody>
</table>
<p>The table above shows a dataframe with protein sequences in it. The second row contains a made up sequence to serve as an example. This made up sequence will be removed with <code>remove_nonstandard_aa()</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">my_clean_protein &lt;-<span class="st"> </span><span class="kw">remove_nonstandard_aa</span>(nonstandard_example_df)</a></code></pre></div>
<table>
<caption>My clean protein</caption>
<thead>
<tr class="header">
<th align="left">seq.name</th>
<th align="left">seq.aa</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G1P6H5_MYOLU</td>
<td align="left">MALTVRIQAACLLLLLLASLTSYSLLLSQTTQLAD</td>
</tr>
</tbody>
</table>
</div>
<div id="step-3-calculate-features-with-calculate_features" class="section level3">
<h3>Step 3: Calculate features with <code>calculate_features()</code></h3>
<p><code>calculate_features()</code> calculates a range of physicochemical properties that are used by the predictive model within <code>predict_AMP_prob()</code> to make its predictions (step 4).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">my_protein_features &lt;-<span class="st"> </span><span class="kw">calculate_features</span>(my_clean_protein)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt; Proteins less than twenty amino acids long were removed and totalled at: 0</span></a></code></pre></div>
<table>
<caption>My protein features ( first six columns )</caption>
<thead>
<tr class="header">
<th align="left">seq.name</th>
<th align="right">Amphiphilicity</th>
<th align="right">Hydrophobicity</th>
<th align="right">pI</th>
<th align="right">Mw</th>
<th align="right">Charge</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G1P6H5_MYOLU</td>
<td align="right">0.415</td>
<td align="right">0.437</td>
<td align="right">8.501</td>
<td align="right">9013.757</td>
<td align="right">4.53</td>
</tr>
</tbody>
</table>
</div>
<div id="step-4-predict-antimicrobial-peptide-probability-with-predict_amp_prob" class="section level3">
<h3>Step 4: Predict antimicrobial peptide probability with <code>predict_AMP_prob()</code></h3>
<p><code>predict_AMP_prob()</code> uses the output from <code>calculate_features()</code> to predict the probability of a protein to be an antimicrobial peptide.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">my_prediction &lt;-<span class="st"> </span><span class="kw">predict_AMP_prob</span>(my_protein_features)</a></code></pre></div>
<table>
<caption>My prediction</caption>
<thead>
<tr class="header">
<th align="left">seq.name</th>
<th align="right">prob_AMP</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G1P6H5_MYOLU</td>
<td align="right">0.8946007</td>
</tr>
</tbody>
</table>
<div id="optional-steps-extract-amp-sequences-and-save-as-fasta-format-file" class="section level4">
<h4>Optional steps: Extract AMP sequences and save as FASTA format file</h4>
<p><code>extract_amps()</code> uses the output from <code>read_faa()</code> and <code>predict_AMP_prob()</code> as parameters to create a new dataframe which contains the sequence name and sequence of the identified antimicrobial peptides at a set probability threshold of &gt;=0.50. The default threshold of &gt;=0.50 can be changed with the “prob” parameter.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">my_predicted_amps &lt;-<span class="st"> </span><span class="kw">extract_amps</span>(<span class="dt">df_w_seq =</span> my_protein, <span class="dt">df_w_prob =</span> my_prediction, <span class="dt">prob =</span> <span class="fl">0.55</span>)</a></code></pre></div>
<table>
<caption>My predicted AMPs</caption>
<thead>
<tr class="header">
<th align="left">seq.name</th>
<th align="left">seq.aa</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G1P6H5_MYOLU</td>
<td align="left">MALTVRIQAACLLLLLLASLTSYSLLLSQTTQLAD</td>
</tr>
</tbody>
</table>
<p><code>df_to_faa()</code> writes a dataframe containing the sequence and corresponding sequence name to a FASTA file.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">df_to_faa</span>(my_predicted_amps, <span class="st">&quot;my_predicted_amps.fasta&quot;</span>)</a></code></pre></div>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
